cpp/src/io/comp/nvcomp_adapter.cpp

```diff
@@ -37,9 +37,11 @@
 
 #define NVCOMP_HAS_DEFLATE(MAJOR, MINOR, PATCH) (MAJOR > 2 or (MAJOR == 2 and MINOR >= 5))
 
-#define NVCOMP_HAS_TEMPSIZE_EX(MAJOR, MINOR, PATCH) \
+#define NVCOMP_HAS_DECOMP_TEMPSIZE_EX(MAJOR, MINOR, PATCH) \
   (MAJOR > 2 or (MAJOR == 2 and MINOR > 3) or (MAJOR == 2 and MINOR == 3 and PATCH >= 1))
 
+#define NVCOMP_HAS_COMP_TEMPSIZE_EX(MAJOR, MINOR, PATCH) (MAJOR > 2 or (MAJOR == 2 and MINOR >= 6))
```

Comment:
So the com and decomp are offered by different nvcomp versions and can be used differently?

Reply:
yes, they were added in different releases (2.3.1 and 2.6).
They can be used independently, there's no interaction between the two API sets.